File: android/lizi/src/main/java/com/ponycui/lizi/Cell.java
// Copyright (c) 2021, PonyCui.
// for details. All rights reserved. Use of this source code is governed by a
// GPLv3 license that can be found in the LICENSE file.

package com.ponycui.lizi;

import android.graphics.Bitmap;

public class Cell {

    public String identifier;

    public Cell(String identifier) {
        this.identifier = identifier;
    }

    public double birthRate = 1;

    public Bitmap contents;

    public double lifttime = 0;
    public double lifttimeRange = 0;

    public double velocity = 0;
    public double velocityRange = 0;

    public double alphaSpeed = 0;
    public double alphaRange = 0;

    public Offset acceleration = new Offset(0,0);

    public double scale = 1.0;
    public double scaleSpeed = 0.0;
    public double scaleRange = 0.0;

    public double emissionLongitude = 0.0;
    public double emissionRange = 0.0;

    public double spin = 0.0;
    public double spinRange = 0.0;

}

File: web/src/cell.ts
// Copyright (c) 2021, PonyCui.
// for details. All rights reserved. Use of this source code is governed by a
// GPLv3 license that can be found in the LICENSE file.

import { Offset } from "./types";

export class Cell {
  constructor(readonly identifier: string) {}

  birthRate: number = 1;

  contents?: HTMLImageElement;

  lifttime: number = 0;
  lifttimeRange: number = 0;

  velocity: number = 0;
  velocityRange: number = 0;

  alphaSpeed: number = 0;
  alphaRange: number = 0;

  acceleration: Offset = { x: 0, y: 0 };

  scale: number = 1.0;
  scaleSpeed: number = 0.0;
  scaleRange: number = 0.0;

  emissionLongitude: number = 0.0;
  emissionRange: number = 0.0;

  spin: number = 0.0;
  spinRange: number = 0.0;
}
